lib/publiccloud/provider.pm

```diff
@@ -337,8 +410,23 @@ Destroys the current terraform deployment
 sub terraform_destroy {
     my ($self) = @_;
     record_info('INFO', 'Removing terraform plan...');
-    assert_script_run('cd ' . TERRAFORM_DIR);
-    script_run('terraform destroy -no-color -auto-approve', TERRAFORM_TIMEOUT);
+    if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
+    }
+    else {
+        assert_script_run('cd ' . TERRAFORM_DIR);
+    }
+    assert_script_run('terraform plan -no-color -destroy -out mydestroy', TERRAFORM_TIMEOUT);
```

Comment:
As the `plan [...] -destroy` way requires to set the variable, I'm not sure is in fact a good idea... Maybe better/easier to let this part as-is, @cfconrad WDYT?

Reply:
Ok for me and I would say, if we discover more problems with leftovers which might be solved by this, we can easy add it later on.

Reply:
I think I will just keep the check and `on_timeout` part, mainly to ensure that resources are correctly destroyed (on Azure!).